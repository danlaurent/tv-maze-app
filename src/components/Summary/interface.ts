export interface ISummary {
  summary: string;
  testID?: string;
}
