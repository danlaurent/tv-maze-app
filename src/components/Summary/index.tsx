import React from 'react';
import { Text, StyleSheet } from 'react-native';
import { GRAY } from '../../constants/styles/colors';
import { cleanSummary } from '../../utils/transform';
import { ISummary } from './interface';

const styles = StyleSheet.create({
  summary: {
    textAlign: 'center',
    color: GRAY,
  },
});

const Summary = ({ summary }: ISummary) => (
  <Text style={styles.summary}>{cleanSummary(summary)}</Text>
);

export default Summary;
