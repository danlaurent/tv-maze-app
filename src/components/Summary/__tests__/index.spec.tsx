import React from 'react';
import { render, RenderAPI } from '@testing-library/react-native';
import Summary from '..';

describe('Summary', () => {
  let tree: RenderAPI;
  beforeEach(() => {
    tree = render(<Summary testID='testSummary' summary='A test summary' />);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
