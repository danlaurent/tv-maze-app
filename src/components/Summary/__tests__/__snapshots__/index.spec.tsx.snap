// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#6d6f7e",
      "textAlign": "center",
    }
  }
  testID="testSummary"
>
  A test summary
</Text>
`;
